### PR TITLE
FIX: Do not error if autogenerated channel slug is blank

### PR DESCRIPTION
**Pull request by martin-brennan**
In certain cases, like when `SiteSetting.slug_generation_method`
is set to `none` with certain locales, the autogenerated chat
channel slugs will end up blank. This was causing errors in
unrelated jobs calling `update!` on the channel. Instead, we
should just copy Category behaviour, which does not error
if the autogenerated slug ends up blank. We already allow
for this with chat channel URLs, using `-` in place of the
missing slug.

### CODE REVIEW

File: plugins/chat/spec/models/category_channel_spec.rb [hunk 1]
```diff
@@ -124,6 +124,13 @@
           channel.validate
           expect(channel.errors.full_messages).to include("Slug is invalid")
         end
+
+        it "does not add a validation error when autogenerated slug ends up empty" do
+          SiteSetting.slug_generation_method = "none"
+          channel.update!(name: "は無効です", slug: nil)
```
Comment (Drenmi) [inline plugins/chat/spec/models/category_channel_spec.rb:130]: Correct me if I'm wrong, but `#update!` would raise an exception on validation error vs `#update` which would add to the `errors`. If so there's a bit of a disconnect between this setup and the assertion on the line below.

File: plugins/chat/spec/models/category_channel_spec.rb [hunk 2]
```diff
@@ -124,6 +124,13 @@
...
+          expect(channel.errors.full_messages).to be_empty
```
Comment (Drenmi) [inline plugins/chat/spec/models/category_channel_spec.rb:131]: Could potentially be `expect(channel).to be_valid`.

**Review by Drenmi — APPROVED** [review]
LGTM! 🚀

File: plugins/chat/spec/models/category_channel_spec.rb (hunk 2 above)
Comment (martin-brennan) [reply plugins/chat/spec/models/category_channel_spec.rb:131]: Ah yes that's better

File: plugins/chat/spec/models/category_channel_spec.rb (hunk 1 above)
Comment (martin-brennan) [reply plugins/chat/spec/models/category_channel_spec.rb:130]: Good point, fixing

**Review by jjaffeux — APPROVED** [review]

**Pull request merged**